Dockerfile

```diff
@@ -4,39 +4,37 @@ FROM golang:1.13 as builder
 ENV CGO_ENABLED=0
 ENV GO111MODULE=on
 
-COPY --from=license-check /license-check /usr/bin/
-
-RUN mkdir -p /go/src/github.com/inlets/inlets-operator
 WORKDIR /go/src/github.com/inlets/inlets-operator
 
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+COPY --from=license-check /license-check /usr/bin/
+
 COPY . .
 
 ARG OPTS
-RUN go mod download
 
 RUN gofmt -l -d $(find . -type f -name '*.go' -not -path "./vendor/*")
 RUN go test -v ./...
+
 RUN VERSION=$(git describe --all --exact-match `git rev-parse HEAD` | grep tags | sed 's/tags\///') && \
   GIT_COMMIT=$(git rev-list -1 HEAD) && \
   env ${OPTS} CGO_ENABLED=0 GOOS=linux go build -ldflags "-s -w \
   -X github.com/inlets/inlets-operator/pkg/version.Release=${VERSION} \
   -X github.com/inlets/inlets-operator/pkg/version.SHA=${GIT_COMMIT}" \
-  -a -installsuffix cgo -o inlets-operator . && \
-  addgroup --system app && \
-  adduser --system --ingroup app app && \
-  mkdir /scratch-tmp
-
-# we can't add user in next stage because it's from scratch
-# ca-certificates and tmp folder are also missing in scratch
-# so we add all of it here and copy files in next stage
-
-FROM scratch
+  -a -installsuffix cgo -o inlets-operator .
 
-COPY --from=builder /etc/passwd /etc/group /etc/
-COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
-COPY --from=builder --chown=app:app /scratch-tmp /tmp/
-COPY --from=builder /go/src/github.com/inlets/inlets-operator/inlets-operator .
+# Use distroless as minimal base image to package the binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
```

Comment:
If this works with multi-arch images then it could be a nice way of making the FROM scratch and copying tmp/passwd/ca-certs cleaner for this build and nats-queue-worker.